### PR TITLE
Add util method to application-data-retrieval-client

### PR DESCRIPTION
**Pull request by kayathiri4**
### Proposed changes in this pull request
**Add following util methods to  application-data-retrieval-client**

- getApplicationID
- getApplicationAccessURLForUUID

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java [hunk 1]
```diff
@@ -96,6 +98,124 @@ public String getApplicationAccessURL(String tenant, String applicationName)
...
 
+    /**
+     * Gets the access url configured for the given application.
+     *
+     * @param tenant tenant domain of the application
+     * @param applicationId UUID of the application
+     * @return the access url configured for the given application
+     * @throws ApplicationDataRetrievalClientException if IO exception occurs or access URL is not configured
+     */
+    public String getApplicationAccessURLForUUID(String tenant, String applicationId)
```
Comment (inthirakumaaran) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java:109]:
```suggestion
    public String getApplicationAccessURLForID(String tenant, String applicationId)
```

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java [hunk 2]
```diff
@@ -96,6 +98,124 @@ public String getApplicationAccessURL(String tenant, String applicationName)
...
+            throws ApplicationDataRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
```
Comment (inthirakumaaran) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java:112]: Cannot we extract this client logic out? let's try to reduce the code duplication.

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java [hunk 3]
```diff
@@ -96,6 +98,124 @@ public String getApplicationAccessURL(String tenant, String applicationName)
...
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new ApplicationDataRetrievalClientException(msg, e);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * Gets the application UUID for the given application.
```
Comment (inthirakumaaran) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java:176]:
```suggestion
     * Gets the application ID for the given application.
```

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java (hunk 1 above)
Comment (inthirakumaaran) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java:109]: Let's refer to this as Application ID

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java (hunk 2 above)
Comment (kayathiri4) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java]: Addressed with 2be0579cdf1d3548b7c3755834647f3a27153d2e

**Review by inthirakumaaran — APPROVED** [review]

**Pull request merged**